Reworked configpanel

pcjco opened the pull request:
The Config panel of the plugin is very messy. I tried to rework it.

- using complete words, phrases to describe all settings instead of abbreviations for all settings (FR/EN)
- Gather all scrapable data choices into a single ListCheckButton instead of the all the CheckButtons
- Gather everything linked to Log/Debug in a third tab
- Replaced the three-state CheckButton for Scraping Web Link into a normal CheckButton (scrape weblink or not) + a new option to store shortened link (compatible with the previous setting storage)
- Replaced the "Pause Scrape" three-state CheckButton (checked=allow user to choose when non univoque choice arise, unchecked=not allowing to choose and always choose 1st item or skipping to next and indeterminate=allow user to choose when non univoque choice arise BUT with a timer...) by a simplified choice "Decision when an ambiguity in the choice of series or book arises" with two options "Ignore and skip to next" or "User choose between choices". To use or not a timer for choice pop-up is now hidden in the TIMEOUT given to this pop-up before it closes (if using "0" = no timer) (compatible with the previous setting storage)
- Changed the max size of log files to show MegaBytes instead of Bytes (stored parameters still in Bytes to keep compatibility)
- Also simplified some regex strings where a lot of characters where protected for no reason (\- \/ \: \")
- Fixed the SERIE_QSERIE_PATTERN that should have returned the name of a serie from an album page (Quickscrape/Rescrape only). Also, the condition to use it was if the serie title was blank ??? I removed this strange condition. A Quickscrape/Rescrape will always reset the Serie Title from the album page info now.

pcjco commented:
![image](https://github.com/user-attachments/assets/25a52db0-30b6-48d0-be46-f45f94ba2665)
![image](https://github.com/user-attachments/assets/c88a6d95-b095-44cd-a0b0-b26220013a13)
![image](https://github.com/user-attachments/assets/8c631ba8-b90b-4d4b-889c-2c7d87dbcfa5)

maforget commented on src/BedethequeScraper2.py:
```diff
@@ -3266,42 +3018,53 @@ def button_Click(self, sender, e):
+        elif sender.Name.CompareTo(self._ButtonCheckNone.Name) == 0:
+            for index in range(self._ScrapedDataCheckedListBox.Items.Count):
+                self._ScrapedDataCheckedListBox.SetItemCheckState(index, CheckState.Unchecked)
+                if index == self._scrapedData.keys().index('Web'):
```
I don't get this part, why is Web Special? Shouldn't it be set by the line above anyway. You are using the same function to set it anyway.

maforget commented on src/BedethequeScraper2.py:
```diff
@@ -3266,42 +3018,53 @@ def button_Click(self, sender, e):
+        elif sender.Name.CompareTo(self._ButtonCheckAll.Name) == 0:
+            for index in range(self._ScrapedDataCheckedListBox.Items.Count):
+                self._ScrapedDataCheckedListBox.SetItemCheckState(index, CheckState.Checked)
+                if index == self._scrapedData.keys().index('Web'):
```
Same here

maforget reviewed (changes requested):
Where did you get your `collections.py` & `heapq.py`? They don't match the files I have at all from IronPython or Python 2.7.18. The collections.py looks like it is from 2.7.3, but even there there is a line that is changed. I've replaced the files from IronPython 2.7.12 locally.

There is something that will not be compatible. In my case if I had the settings to keep the window open and wait (CBStop to true), and there was a timeout value like is the default. In this case someone would need to go into the setting and change the timeout to 0 for it to stay open. So anyone updating without knowing would have a different behavior. Also the relation between both setting isn't clear, the value isn't in the group box and there is no indication that you need to set it to 0 to keep the window open.

maforget commented:
It does look nicer and the choice are more clear, even for me. I never even knew that Web had 3 state and the short links

Please review my comments in the review. 

What program did you use to redo the UI? Did you do it by hand or are you using Sharp Develop?

Also if possible commits should only be for 1 thing. For example removing the \ in the regex could have been separate. It is easier to track the changes and easier to revert if required.

I will get feedback from a discord group that I am a member regarding your changes.

maforget commented on src/BedethequeScraper2.py:
```diff
@@ -2778,343 +2490,387 @@ def __init__(self):
         self._DBGLOGMAX.TabIndex = 4
-        self._DBGLOGMAX.Text = str(DBGLOGMAX)
+        self._DBGLOGMAX.TextAlign = HorizontalAlignment.Center
+        self._DBGLOGMAX.Text = str(int(DBGLOGMAX / 1024 / 1024))
```
This is a breaking change. The default value is 10 000 bytes, so around 10KiB. So your default value will be 0.009 Mb, so since it's an int it will changed to 0. So on opening the setting window it will save the value to 0, So on each and every run it will ask to replace the files. 

We can't change the default value, because existing settings wouldn't not work either. 
 * We can either change it to a double/float
 * Just change it to Kb instead
 * But a floor to be 1 minimum, so it would be 10KB if you have an existing setting file, and at the same time change the minimum to at least 1MB.

pcjco replied:
I choose to use floating value representing MegaBytes in UI and then store Bytes to fix that.

maforget commented on src/BedethequeScraper2.py:
```diff
@@ -2778,343 +2490,387 @@ def __init__(self):
         self._RENLOGMAX.TabIndex = 5
-        self._RENLOGMAX.Text = str(RENLOGMAX)
+        self._RENLOGMAX.TextAlign = HorizontalAlignment.Center
+        self._RENLOGMAX.Text = str(int(RENLOGMAX / 1024 / 1024))
```
Exact same problem than DBGLOGMAX, it will be 0.

maforget commented on src/BedethequeScraper2.py:
```diff
@@ -3174,52 +2935,43 @@ def __init__(self):
             DBGONOFF = if_else(self._DBGONOFF.CheckState == CheckState.Checked, True, False)
-            DBGLOGMAX = int(self._DBGLOGMAX.Text)
-            RENLOGMAX = int(self._RENLOGMAX.Text)
+            DBGLOGMAX = int(self._DBGLOGMAX.Text)*1024*1024
```
breaking change see above

maforget commented on src/BedethequeScraper2.py:
```diff
@@ -3174,52 +2935,43 @@ def __init__(self):
-            DBGLOGMAX = int(self._DBGLOGMAX.Text)
-            RENLOGMAX = int(self._RENLOGMAX.Text)
+            DBGLOGMAX = int(self._DBGLOGMAX.Text)*1024*1024
+            RENLOGMAX = int(self._RENLOGMAX.Text)*1024*1024
```
breaking change see above

maforget reviewed (changes requested):
There will be a problem with the change to MB in the Logs

maforget commented on src/BedethequeScraper2.py:
```diff
@@ -3266,42 +3018,53 @@ def button_Click(self, sender, e):
+        elif sender.Name.CompareTo(self._ButtonCheckNone.Name) == 0:
+            for index in range(self._ScrapedDataCheckedListBox.Items.Count):
+                self._ScrapedDataCheckedListBox.SetItemCheckState(index, CheckState.Unchecked)
+                if index == self._scrapedData.keys().index('Web'):
```
I don't get this part, why is Web Special? Shouldn't it be set by the line above anyway. You are using the same function to set it anyway.

pcjco replied:
Fixed.

maforget commented on src/BedethequeScraper2.py:
```diff
@@ -3266,42 +3018,53 @@ def button_Click(self, sender, e):
+        elif sender.Name.CompareTo(self._ButtonCheckAll.Name) == 0:
+            for index in range(self._ScrapedDataCheckedListBox.Items.Count):
+                self._ScrapedDataCheckedListBox.SetItemCheckState(index, CheckState.Checked)
+                if index == self._scrapedData.keys().index('Web'):
```
Same as above. why is Web Special?

pcjco replied:
Fixed.

pcjco commented:
> Where did you get your `collections.py` & `heapq.py`? They don't match the files I have at all from IronPython or Python 2.7.18. The collections.py looks like it is from 2.7.3, but even there there is a line that is changed. I've replaced the files from IronPython 2.7.12 locally.
 
It seems the IronPython is 2.7.4. That is that version used as Package Reference in ComicRackCE/ComicRack.Plugins/ComicRack.Plugins.csproj
```
  <PackageReference Include="IronPython">
      <Version>2.7.4</Version>
    </PackageReference>
```

So I downloaded it from https://github.com/IronLanguages/main/releases/tag/ipy-2.7.4 and looked there for the collections.py (that need heapq.py)

> There is something that will not be compatible. In my case if I had the settings to keep the window open and wait (CBStop to true), and there was a timeout value like is the default. In this case someone would need to go into the setting and change the timeout to 0 for it to stay open. So anyone updating without knowing would have a different behavior. Also the relation between both setting isn't clear, the value isn't in the group box and there is no indication that you need to set it to 0 to keep the window open.

I changed the behavior to keep compatibility.
![image](https://github.com/user-attachments/assets/ce7973aa-81dc-4070-b4a9-44d55840327d)

> What program did you use to redo the UI? Did you do it by hand or are you using Sharp Develop?

By hand.
